x/genutil/types/genesis_state.go

```diff
@@ -100,15 +100,15 @@ func GenesisStateFromGenFile(cdc codec.JSONMarshaler, genFile string) (genesisSt
 // ValidateGenesis validates GenTx transactions
 func ValidateGenesis(genesisState GenesisState) error {
 	for i, genTx := range genesisState.GenTxs {
-		var tx authtypes.StdTx
+		var tx sdk.Tx
 		if err := ModuleCdc.UnmarshalJSON(genTx, &tx); err != nil {
```

Comment:
This should use `TxJSONDecoder`

Reply:
I ended up adding a parameter to `ValidateGenesis`. We can't really add it to `genutil` `AppModuleBasic` because the `TxConfig` depends on registering all interfaces which requires the `AppModuleBasic`s so that creates a cyclic dependency.